CriticalMass/UIViewController+Management.swift

```diff
@@ -9,17 +9,40 @@
 import UIKit
 
 extension UIViewController {
-    func add(child viewController: UIViewController) {
-        addChild(viewController)
-        view.addSubview(viewController.view)
-        viewController.view.frame = view.bounds
-        viewController.view.autoresizingMask = [.flexibleWidth, .flexibleHeight]
-        viewController.didMove(toParent: self)
+    func add(_ child: UIViewController) {
+        addChild(child)
+        view.addSubview(child.view)
+        child.didMove(toParent: self)
     }
 
-    func remove(child viewController: UIViewController) {
-        viewController.willMove(toParent: nil)
-        viewController.view.removeFromSuperview()
-        viewController.removeFromParent()
+    func remove() {
+        // Just to be safe, we check that this view controller
+        // is actually added to a parent before removing it.
+        guard parent != nil else {
+            return
```

Comment:
Maybe worth logging a warning as this shouldn’t happen

Reply:
I could add
```swift
Logger.log(.debug, log: .default, "Expected to have parent viewController")
```

Reply:
sounds good. Maybe we could also add the ViewControllers type to the message so it's easier to find